packages/inertia-react/index.d.ts

```diff
@@ -86,3 +86,29 @@ export interface InertiaFormProps<TForm = Record<string, any>> {
 }
 
 export function useForm<TForm = Record<string, any>>(initialValues: TForm): InertiaFormProps<TForm>;
+
+export interface AppSetupPageProps {
+    initialPage: Inertia.PageProps,
+    initialComponent: any,
+    resolveComponent: (name: string) => Promise<any>,
```

Comment:
```suggestion
    resolveComponent: (name: string) => React.ComponentType | Promise<React.ComponentType>,
```
See https://github.com/inertiajs/inertia/blob/master/packages/inertia-react/index.d.ts#L16